crates/compression/src/tables.rs

```diff
@@ -0,0 +1,210 @@
+use fuel_core_types::{
+    fuel_compression::{
+        CompressibleBy,
+        DecompressibleBy,
+        RegistryKey,
+    },
+    fuel_tx::{
+        input::PredicateCode,
+        Address,
+        AssetId,
+        ContractId,
+        ScriptCode,
+    },
+};
+use std::collections::HashMap;
+
+use crate::{
+    compress::{
+        CompressCtx,
+        CompressDb,
+        PrepareCtx,
+    },
+    decompress::{
+        DecompressCtx,
+        DecompressDb,
+        DecompressError,
+    },
+    ports::{
+        EvictorDb,
+        TemporalRegistry,
+    },
+};
+
+macro_rules! tables {
+    ($($name:ident: $type:ty),*$(,)?) => {
+        #[doc = "RegistryKey namespaces"]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+        #[allow(non_camel_case_types)] // Match names in structs exactly
+        pub enum RegistryKeyspace {
+            $(
+                $name,
+            )*
+        }
+
+        #[doc = "RegistryKey namespace with an associated typed value"]
+        #[derive(Debug, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+        #[allow(non_camel_case_types)] // Match names in structs exactly
+        pub enum RegistryKeyspaceValue {
```

Comment:
I prefer if we define this type outside the macros since it is not related to compression and belongs to the storage part

Reply:
I just removed it in 7fe125fdfe3a48e1f9de1dd400027ee2d06d66e4